pkg/config/config.go

```diff
@@ -271,19 +272,37 @@ func (c *Config) ShouldGenerateInterface(ctx context.Context, packageName, inter
 	}
 	_, interfaceExists := interfacesSection[interfaceName]
 
-	var matchedByRegex bool
+	matchedByRegex := false
 	if pkgConfig.IncludeRegex != "" {
 		if pkgConfig.All {
 			log := zerolog.Ctx(ctx)
-			log.Warn().Msg("interface config has both `all` and `include-regex` set. `include-regex` will be ignored")
+			log.Warn().Msg("interface config has both `all` and `include-regex` set: `include-regex` will be ignored")
 		} else {
 			matchedByRegex, err = regexp.MatchString(pkgConfig.IncludeRegex, interfaceName)
 			if err != nil {
-				return false, err
+				return false, fmt.Errorf("evaluating `include-regex`: %w", err)
 			}
 		}
 	}
-	return pkgConfig.All || interfaceExists || matchedByRegex, nil
+	excludedByRegex := false
+	if pkgConfig.ExcludeRegex != "" {
+		if pkgConfig.All {
+			log := zerolog.Ctx(ctx)
+			log.Warn().Msg("interface config has both `all` and `exclude-regex` set: `exclude-regex` will be ignored")
+		} else if pkgConfig.IncludeRegex == "" {
+			log := zerolog.Ctx(ctx)
+			log.Warn().Msg("interface config has `exclude-regex` set but not `include-regex`: `exclude-regex` will be ignored")
+		} else {
+			excludedByRegex, err = regexp.MatchString(pkgConfig.ExcludeRegex, interfaceName)
+			if err != nil {
+				return false, fmt.Errorf("evaluating `exclude-regex`: %w", err)
+			}
+			if excludedByRegex {
+				matchedByRegex = false
+			}
```

Comment:
I would say we can reorganize this code. We should have a separate section that logs the warnings like:

```go
if pkgConfig.All {
    if pkgConfig.IncludeRegex != "" {
        log.Warn().Msg("interface config has both `all` and `include-regex` set: `include-regex` will be ignored")
    }
    if pkgConfig.ExcludeRegex != "" {
        log.Warn().Msg("interface config has both `all` and `exclude-regex` set: `exclude-regex` will be ignored")
    }
    return true
}



```

Then populate the `matchedBy`/`excludedBy` variables if `pkgConfig.All == true`

```go
var matchedByRegex, excludedByRegex bool
if pkgConfig.IncludeRegex != "" {
    // do matching
}
if pkgConfig.ExcludeRegex != "" {
    // do matching
}
return interfaceExists || (matchedByRegex && !excludedByRegex)
```

Reply:
I have taken a few liberties to reduce nesting, LMK if it needs more work